### PR TITLE
Feat/new managers

### PR DESCRIPTION
**Pull request by Eein**
- Adds CurrencyManager
- Adds LevelManager
- Add Helper information for above

### CODE REVIEW

File: src/gui/helpers/inventory_helper.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+impl GuiHelper for InventoryHelper {
+    fn draw(
+        &mut self,
+        game_state: &GameState,
+        _sequencer: &mut Sequencer<GameState>,
+        ui: &mut egui::Ui,
+        _tab: &mut String,
+    ) {
+        let currency_manager = &game_state.memory_managers.currency_manager.data;
+
```
Comment (orkaboy) [inline src/gui/helpers/inventory_helper.rs:25]: Possibly mark this with a debug printout "more to be implemented" (assuming the actual inventory will go here later)

File: src/memory/currency_manager.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+use crate::memory::memory_context::MemoryContext;
+use crate::memory::{MemoryManager, MemoryManagerUpdate};
+use crate::state::StateContext;
+use log::info;
+use memory::memory_manager::il2cpp::UnityMemoryManager;
+use memory::process::MemoryError;
+
+impl Default for MemoryManager<CurrencyManagerData> {
```
Comment (orkaboy) [inline src/memory/currency_manager.rs:8]: I'm always a bit jarred when the `impl` comes before the actual `struct`, though I appreciate Rust for allowing it.

File: src/memory/level_manager.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                Err(_) => {
+                    self.scene_name = "None".to_string();
+                }
+            };
+        }
+        Ok(())
+    }
+
+    pub fn update_scene_guid(&mut self, memory_context: &MemoryContext) -> Result<(), MemoryError> {
+        if let Ok(addr) = memory_context.follow_fields::<u64>(&["currentLevel"]) {
```
Comment (orkaboy) [inline src/memory/level_manager.rs:72]: Is there a valid list of what this (and `mainSceneName` above) can be somewhere? Or do we have to compile one ourselves if we want to use it?

**Review by orkaboy — APPROVED** [review]

File: src/memory/level_manager.rs (hunk 3 above)
Comment (Eein) [reply src/memory/level_manager.rs:72]: We'll have to compile it ourselves unfortunately - i can try dumping from game files if I haven't already and forgotten.

File: src/memory/currency_manager.rs (hunk 2 above)
Comment (Eein) [reply src/memory/currency_manager.rs:8]: Happy to make a cleanup ticket and knock this out tom - I agree as a preference but i dont pay attention

**Pull request merged**